subsys/secure_storage/Kconfig

```diff
@@ -0,0 +1,175 @@
+#
+# Copyright (c) 2018 - 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Secure storage subsystem configuration options
+
+menuconfig SECURE_STORAGE
+	bool "Secure Storage"
+	depends on !BUILD_WITH_TFM
+	help
+	  The secure storage subsystem allows its users to store in a secure
+	  way data ensuring data integrity and confidentially by using AEAD
+	  algorithms. It supports several secure implementation back-ends to
+	  provide various levels of trust depending on the device security
+	  features.
+
+if SECURE_STORAGE
+module = SECURE_STORAGE
+module-str = secure_storage
+source "subsys/logging/Kconfig.template.log_config"
+
+
+menuconfig PSA_PROTECTED_STORAGE
+	bool "Protected Storage"
+	depends on SECURE_STORAGE
+	help
+	  The Protected Storage is designed to store in a secure
+	  way device intimate data and protects against physical attacks.
+	  It supports several storage back-ends and several secure
+	  implementation back-ends to provide various levels of trust.
+
+if PSA_PROTECTED_STORAGE
+
+config PSA_PROTECTED_STORAGE_PREFIX
+	string "Protected Storage object prefix"
+	depends on PSA_PROTECTED_STORAGE
+	default "ps/"
+	help
+	  This defines the prefix for Protected Storage object names.
+
+endif # PSA_PROTECTED_STORAGE
+
+menuconfig PSA_INTERNAL_TRUSTED_STORAGE
+	bool "Internal Trusted Storage"
+	depends on SECURE_STORAGE
+	default y
+	help
+	  The Internal Trusted Storage is designed to store assets that must
+	  be placed inside internal flash. Some examples of assets that require
+	  this are replay protection values for external storage and keys for
+	  use by components of the PSA Root of Trust.
+
+if PSA_INTERNAL_TRUSTED_STORAGE
+
+config PSA_INTERNAL_TRUSTED_STORAGE_PREFIX
+	string "Internal Trusted Storage object prefix"
+	depends on PSA_INTERNAL_TRUSTED_STORAGE
+	default "its/"
+	help
+	  This defines the prefix for Internal Trusted Storage object names.
+
+endif # PSA_INTERNAL_TRUSTED_STORAGE
+
+choice SECURE_STORAGE_BACKEND
+	prompt "Trust backend"
+	depends on SECURE_STORAGE
+	default SECURE_STORAGE_AEAD_BACKEND
+	help
+	  The backend that handles encryption, authentication or other means of
+	  trust validation of the stored assets.
+
+menuconfig SECURE_STORAGE_AEAD_BACKEND
+	bool "Use AEAD scheme"
+	help
+	  Use an AEAD scheme to encrypt the asset data and authenticate its
+	  data including the meta data.
+
+if SECURE_STORAGE_AEAD_BACKEND
+
+config SECURE_STORAGE_AEAD_BACKEND_MAX_DATA_SIZE
+	int "AEAD backend maximum storage size"
+	depends on SECURE_STORAGE_AEAD_BACKEND
+	default 256
+	help
+	  This defines the maximum data size that can be stored.
+
+choice SECURE_STORAGE_AEAD_AEAD_CRYPTO_BACKEND
```

Comment:
AEAD is twice here.
~~Suggest using ALG instead of BACKEND.~~
BACKEND should be before AEAD (see above)